WordPress/openverse-frontend
# Add Content switcher and Page menu buttons

## Fixes


Related to #485 by @zackkrida

## Description


This PR adds the buttons for the Content Switcher (the button that allows the user to select the content type to search) and the Page menu (with links to content pages on Openverse). It also updates the button styles to use `tertiary` for the button that has a border, and `action-menu` for the button that has no border. Another change is to set the `pressed` style based on the `aria-pressed` or `aria-expanded` properties.
This change was extracted from a larger Content Switcher [draft PR](https://github.com/WordPress/openverse-frontend/pull/490) based on Krysal's [comment](https://github.com/WordPress/openverse-frontend/pull/490#issuecomment-1000562201) in a larger Content Switcher [draft PR](https://github.com/WordPress/openverse-frontend/pull/490).

To prevent the need for props drilling, it also uses `provide` in the VHeader component to provide all child components with values of `sHeaderScrolled` and `isMdScreen`.


## Testing Instructions

Run the project using `pnpm run dev`, and go to `localhost:8443/search/images/?q=cat`. You should see two buttons in the header.
#### Desktop, header not scrolled
<img width="768" alt="Screen Shot 2022-01-03 at 9 04 02 PM" src="https://user-images.githubusercontent.com/15233243/147964031-5a430142-69f5-4030-a37b-866fda6509ed.png">

#### Desktop, header scrolled
<img width="766" alt="Screen Shot 2022-01-03 at 9 03 50 PM" src="https://user-images.githubusercontent.com/15233243/147964035-36907063-5ae1-4d7f-b105-f04cc3d38c52.png">

#### Mobile, header not scrolled
<img width="487" alt="Screen Shot 2022-01-03 at 9 06 57 PM" src="https://user-images.githubusercontent.com/15233243/147964414-0f230e33-8afd-4248-8cde-54b704660164.png">


#### Mobile, header scrolled
<img width="440" alt="Screen Shot 2022-01-03 at 9 03 31 PM" src="https://user-images.githubusercontent.com/15233243/147964038-d2be9160-3e6e-4aea-8cd5-a87a34827573.png">

Please note, that the buttons are not functioning yet, and their placement is not correct.
## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
